imxrt-hal/src/dma/element.rs

```diff
@@ -0,0 +1,36 @@
+//! Trait to generalize acceptable DMA transfer elements
+
+/// Describes a transferrable DMA element; basically, an unsigned
+/// integer of any size.
+pub trait Element: private::Sealed {
+    /// An identifier describing the data transfer size
+    ///
+    /// Part of the TCD API; see documentation on TCD[SSIZE]
+    /// and TCD[DSIZE] for more information.
+    const DATA_TRANSFER_ID: u16;
+}
+
+impl Element for u8 {
+    const DATA_TRANSFER_ID: u16 = 0;
+}
+
+impl Element for u16 {
+    const DATA_TRANSFER_ID: u16 = 1;
+}
+
+impl Element for u32 {
+    const DATA_TRANSFER_ID: u16 = 2;
+}
+
+impl Element for u64 {
+    const DATA_TRANSFER_ID: u16 = 3;
+}
+
```

Comment:
I suppose to support the 32-byte burst feature here something perhaps a little more special might be needed? Would likely need to be setup to look like 

```impl Element for u8[32]```
or
```impl Element for u16[16]```

And so on down the line of uint types I'd guess?

Reply:
The `SSIZE` documentation in the reference manual says that the burst is "4 beats of 64 bits." I'd be partial to _only_ support `[u64; 4]`, as it describes the "beats" of the burst.

Am I misinterpreting the 32-byte burst?